sensor/kubernetes/listener/resources/secrets.go

```diff
@@ -265,78 +261,51 @@ func (s *secretDispatcher) processDockerConfigEvent(secret, oldSecret *v1.Secret
 		}
 	}
 
-	// In Kubernetes, the `default` service account always exists in each namespace (it is recreated upon deletion).
-	// The default service account always contains an API token.
-	// In OpenShift, the default service account also contains credentials for the
-	// OpenShift Container Registry, which is an internal image registry.
-	fromDefaultSA := saName == defaultSA
-	isGlobalPullSecret := secret.GetNamespace() == openshiftConfigNamespace && secret.GetName() == openshiftConfigPullSecret
+	s.processSecretForLocalScanning(secret, action, dockerConfig, saName)
 
 	newIntegrationSet := set.NewStringSet()
 	for registryAddress, dce := range dockerConfig {
```

Comment:
This does seem a lot more readable now, thank you. I haven't dove too deep into mapping what exactly was done previously vs now because now things are in different files, but I can do that verification, if desired

Reply:
I've done it a few times myself - but would appreciate another set of eyes.  The existing unit tests passed after the modifications which was a good sign - I had compared the old (on master) `processDockerConfigEvent` method to the new (this PR), and ignored any 'logic' specific to delegated scanning.  From my eyes (+ unit tests and e2e tests) the same behavior is achieved.